File: SesionC++/AtributosComoObjeto/ArchivosDeOrigen/Biblioteca.cpp
#include "../ArchivosDeEncabezado/Biblioteca.h"
#include <string>
#include <iostream>
using namespace std;

Autor::Autor(string nombre, string nacionalidad){
    this->nombre = nombre;
    this->nacionalidad = nacionalidad;
}

string Autor::toString(){
    return "Nombre: " + this->nombre + "\n" + "Nacionalidad: " + this->nacionalidad;
}

Libro::Libro(string titulo, int anioPublicacion, Autor autor) : autor(autor) {
    this->titulo = titulo;
    this->anioPublicacion = anioPublicacion;
}

string Libro::toString(){
    return autor.toString() + "\nTitulo: " + this->titulo + "\nAño de publicacion: " + to_string(this->anioPublicacion);
}

File: SesionPy/AtributosComoObjeto/Biblioteca.py
class Autor:
    def __init__(self, nombre, nacionalidad):
        self.nombre = nombre
        self.nacionalidad = nacionalidad

    def __str__(self):
        return f"Nombre: {self.nombre}\nNacionalidad: {self.nacionalidad}"

class Libro:
    def __init__(self, titulo, anioPublicacion, autor):
        self.titulo = titulo
        self.anioPublicacion = anioPublicacion
        self.autor = autor

    def __str__(self):
        return f"Titulo: {self.titulo}\nAño de publicacion: {self.anioPublicacion}\n{self.autor}"